scalding-commons-macros/src/test/scala/com/twitter/scalding/commons/macros/MacrosUnitTests.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ Copyright 2014 Twitter, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+package com.twitter.scalding.commons.macros
+
+import org.scalatest.{ Matchers, WordSpec }
+
+import com.twitter.scalding._
+import com.twitter.scalding.macros._
+import com.twitter.scalding.commons.macros._
+import com.twitter.scalding.serialization.Externalizer
+import com.twitter.scalding.typed.OrderedBufferable
+import com.twitter.bijection.macros.{ IsCaseClass, MacroGenerated }
+import com.twitter.scalding.commons.thrift.TBaseOrderedBufferable
+import com.twitter.scalding.commons.macros.impl.TBaseOrderedBufferableImpl
+import scala.language.experimental.macros
+import org.apache.thrift.TBase
+import java.nio.ByteBuffer
+import scala.util.Success
+
+class MacrosUnitTests extends WordSpec with Matchers {
+  implicit def toTBaseOrderedBufferable[T <: TBase[_, _]]: TBaseOrderedBufferable[T] = macro TBaseOrderedBufferableImpl[T]
+
+  def isMg[T](t: T): T = {
+    t shouldBe a[MacroGenerated]
+    t
+  }
+
+  private val dummy = new TBaseOrderedBufferable[Nothing] {
+    override val minFieldId: Short = 1
+    @transient lazy val prototype: Nothing = null.asInstanceOf[Nothing]
+    override val classz = classOf[Nothing]
+  }
+
+  def rt[T <: TBase[_, _]](t: T)(implicit orderedBuffer: TBaseOrderedBufferable[T]) {
+    val buf = serialize[T](t)
+    assert(orderedBuffer.get(buf).map(_._2) === Success(t))
+  }
+
+  def serialize[T <: TBase[_, _]](t: T)(implicit orderedBuffer: TBaseOrderedBufferable[T]): ByteBuffer = {
```

Comment:
this does not need to take TBaseOrdered... just OrderedBufferable is enough. (same above).